homework-submission/week2/App.js

```diff
@@ -0,0 +1,127 @@
+const List = ({ description, deadline, className, removeFromState }) => {
+  return (
+    <li className={className} onClick={removeFromState}>
+      {description}, {deadline}
+    </li>
+  );
+};
+
+const UnorderedDynamicList = ({ todos, removeFromState }) => {
+  const processTodos = todos.map(todo => (
+    <List
+      description={todo.description}
+      deadline={todo.deadline}
+      key={todo.id}
+      className={todo.done ? 'finished' : ''}
+      removeFromState={() => {
+        removeFromState(todo.id);
+      }}
+    />
+  ));
+
+  return <ul>{processTodos}</ul>;
+};
+
+class DynamicList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      toDos: [
+        {
+          id: 1,
+          description: 'Get out of bed',
+          deadline: '2017-09-11',
+          done: true,
+        },
+        {
+          id: 2,
+          description: 'Brush teeth',
+          deadline: '2017-09-10',
+          done: false,
+        },
+        {
+          id: 3,
+          description: 'Eat breakfast',
+          deadline: '2017-09-09',
+          done: false,
+        },
+      ],
+      newItem: {
+        id: '',
+        description: 'Description',
+        deadline: 'nn',
+        done: false,
+      },
+    };
+  }
+
+  NewListItemDescription = () => {
```

Comment:
I see double,
starting from line 58 ending with line 74, is the exact same as starting from line 76 ending with line 92.

The only difference in these two components is one word in line 68 vs line 86 _(description)_ changed to _(deadline)_.

In any kind like of this situation, remember to dry **(Don't repeat yourself)**
to solve this you can do the following:
```javascript
   NewListItem = (props) => {
    return (
      <p>
        <label htmlFor="">Deadline</label>
        <input
          type="datetime-local"
          name={props.name} // since this is a component we can specify the name via props
          onChange={event => {
            const inputValue = event.target.value;

            this.setState({
              newItem: {
                 ...this.state.newItem,
                 // using the name in this kind of use allowing use to modify the key name to be as we specify the value of it
                 [props.name]: inputValue,
                 id: Date.now()
              },
            });
          }}
        />
      </p>
    );
  };
```
Example use of changed key name:
 name is 'description' means our object will look like: {..., description: 'somevalue'}
 same if 'deadline' means our object will look like: {..., deadline: 'some deadline'}

**Don't forgot to pass the name through props**